content/backend/typescript-apollo/1-getting-started.md

```diff
@@ -0,0 +1,464 @@
+
+In your terminal, go to the directory you created previously and install `typescript` and `ts-node-dev` as dev dependencies. You will also create a `tsconfig.json` file.
+
+```bash(path=".../hackernews-typescript/")
+npm install --save-dev typescript@^4.3.5 ts-node-dev@^1.1.8
+touch tsconfig.json              
+```
+
+</Instruction>
+
+> **Note**: The above code block is annotated with a directory name. It indicates where you need to execute the terminal command.
+
+Great! Now that you have an empty `tsconfig` file, you need to specify the configuration options. For this project, you will use fairly standard options.
+
+<Instruction> 
+
+Copy paste the configuration to your `tsconfig.json`
+
+```json(path=".../hackernews-typescript/tsconfig.json")
+{
+  "compilerOptions": {
+    "target": "ES2018",
+    "module": "commonjs",
+    "lib": [
+      "esnext"
+    ],
+    "strict": true,
+    "rootDir": ".",
+    "outDir": "dist",
+    "sourceMap": true,
+    "esModuleInterop": true
+  }
+}
+```
+</Instruction>
+
+> **Note:** Setting `"strict": true` enables a wide range of type checking behavior that results in more type-safe programs. However it can also lead to some type errors that are actually rather benign. If you get stuck with a type error that you know isn't a bug, but you are having difficulty satisfying the typescript compiler, you could consider turning this to `false` or using the `// @ts-ignore` comment to turn off that specific error/warning. 
+
+
+To learn more about the options used here, you can check out the [tsconfig reference](https://www.typescriptlang.org/tsconfig) in the Typescript documentation. 
+
+### Creating your server layout
+
+
+Before writing any code, you will need to install some dependencies. You will need `nexus` for generating the GraphQL schema. You will also need to install `graphql` and `apollo-server` for creating an HTTP web server with GraphQL capabilities. 
+
+<Instruction>
+
+Install the necessary packages along with their type definitions
+
+```bash
+npm install apollo-server@^3.1.1 graphql@^15.5.1 nexus@^1.1.0
+```
+
+</Instruction>
+
+[`apollo-server`](https://github.com/apollographql/apollo-server/tree/main/packages/apollo-server) is a fully-featured
+GraphQL server. It is based on [Express.js](https://expressjs.com/) and a few other libraries to help you build
+production-ready GraphQL servers.
+
+Here's a list of some cool features it has:
+
+- GraphQL spec-compliant
+- Out-of-the-box support for two GraphQL clients: Apollo Explorer and GraphQL Playground 
+- Can be configured for use with an [Express](https://expressjs.com/) server. 
+- Query performance tracing
+- Runs everywhere: Can be deployed via Vercel, Standard VMs, AWS Lambda, Heroku etc.
+
+[`nexus`](https://github.com/graphql-nexus/nexus) is a library to create Type-Safe GraphQL Schemas with a Code-first approach (Since you write standard javascript/typescript code to define what your schema will look like, hence "code-first"). 
```

Comment:
```suggestion
[`nexus`](https://github.com/graphql-nexus/nexus) is a library to create type-safe GraphQL schemas with a [code-first](https://www.prisma.io/blog/the-problems-of-schema-first-graphql-development-x1mn4cb0tyl3) approach (since you write standard javascript/typescript code to define what your schema will look like, hence "code-first"). 
```